share/trick/trickops/TrickWorkflow.py

```diff
@@ -100,20 +177,112 @@ def __init__(self, project_top_level, log_dir, trick_dir, config_file, cpus=3, q
+        self.config_errors =  []        # Contains errors in setup of management classes
         self.compare_errors = False     # True if comparison errors were found
         self.sims = []                  # list of Sim() instances, filled out from config file
         self.config_file = config_file  # path to yml config
         self.cpus = cpus                # Number of CPUs this workflow should use when running
-        # 'loose' or 'strict' controls if multiple input files per RUN dir are allowed in YAML config file
-        self.parallel_safety = 'loose'
-        self.config = self._read_config(self.config_file) # Contains resulting Dict parsed by yaml.load
+        self.yaml_verifier = TrickWorkflowYamlVerifier(self.config_file)
+        # If not found in the config file, these defaults are used
         self.trick_dir = trick_dir
         self.trick_host_cpu = self.get_trick_host_cpu()
-        self._validate_config()
+        self.env = ''
+        self.config = self.yaml_verifier.verify()
+        self.parsing_errors = self.yaml_verifier.parsing_errors  # All errors found during parsing
+        for e in self.parsing_errors:
+            tprint(e, 'DARK_RED')
+        self._populate_sims()
+
+    def _populate_sims(self):
+        """
+        Given a verified self.config, populate the self.sims list of Sim() instances
+        Note that all the checking of expected dict key/values has already taken place
+        during yaml verification, so we can safely access all dict keys here.
+        Furthermore, any sim or run without required key/values have already been purged
+        from self.config so we can create all management classes at the sim and run
+        levels without worry.
+        """
+        def cprint(msg, color):
+            self.config_errors.append(msg)
+            tprint(msg, color)
+        self.env = self.config['globals']['env']
+        all_sim_paths = [] # Keep a list of all paths for uniqueness check
+        for s in self.config:
+          if not str(s).startswith('SIM'):  # Ignore everything not starting with SIM
+            continue
+          if (not os.path.exists(os.path.join(self.project_top_level, self.config[s]['path'])) ):
+            cprint("ERROR: %s's 'path' %s not found. Continuing but skipping this entire entry from %s."
+                 % (s, self.config[s]['path'], self.config_file), 'DARK_RED')
+            continue
+          if self.config[s]['path'] in all_sim_paths:
+            cprint("ERROR: %s's 'path' is not unique in %s. Continuing but ignoring this sim."
+                % (s, self.config_file), 'DARK_RED')
+            continue
+          # Add the CPU format string (won't fail if there is no CPU placeholder)
+          self.config[s]['binary'] = self.config[s]['binary'].format(cpu=self.trick_host_cpu)
+          # Add the full path to the build command
+          self.config[s]['build_command'] = os.path.join(self.trick_dir, "bin", self.config[s]['build_command'])
```

Comment:
Hey @jdeans289 I just came across a potential use of `build_command` that this isn't compatible with. One of my gateway users is trying to do this:
![Screen Shot 2023-01-20 at 9 28 01 AM](https://user-images.githubusercontent.com/3249339/213736139-81160629-bc2d-461f-8f9c-08b6d7b8654d.png)
I think we can pretty easily support this by doing a `str.replace('trick-CP', os.path.join(self.trick_dir, "bin/trick-CP"))`, but I guess the question I have is whether or not you think this should be allowed. A few weeks ago we asked out loud if there was any use case beyond flags passed to `trick-CP`. Well here's one - setting an environment variable that should only apply to one sim's build.  What do you think?

Reply:
I've replace `build_command` with `build_args` per discussion today. Simpler and less prone to error 👍